libs/platform/i18n/src/lib/locale/default-locale.service.ts

```diff
@@ -57,34 +57,35 @@ export class DefaultLocaleService implements LocaleService {
     this.setActive$.next(value);
   }
 
-  formatDate(
-    stamp: string | number | Date,
-    showTime = false,
-    showDate = false
-  ): Observable<string> {
-    let template = {};
-
-    if (showDate) {
-      template = {
-        day: 'numeric',
-        month: 'short',
-        year: 'numeric',
-      };
-    }
+  formatDate(stamp: string | number): Observable<string> {
+    return this.get().pipe(
+      map((locale) =>
+        Intl.DateTimeFormat(locale.replace('_', '-')).format(new Date(stamp))
```

Comment:
I think now it's possible to cache each of the Intl instance for specific method and reuse it between the calls to optimize formatting performance.
You can create 3 observables which will compute it's own specific Intl instance with shareReplay() operators so that it's not recomputed for every call.